packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1042,9 +1062,58 @@ RawFloatingCursorPoint _toTextPoint(FloatingCursorDragState state, Map<String, d
 
 /// An low-level interface to the system's text input control.
 ///
+/// To start interacting with the system's text input control, call [attach] to
+/// establish a [TextInputConnection] between the system's text input control
+/// and a [TextInputClient]. The majority of commands available for
+/// interacting with the text input control reside in the returned
+/// [TextInputConnection]. The communication between the system text input and
+/// the [TextInputClient] is asynchronous.
+///
+/// The platform text input plugin (which represents the system's text input)
+/// and the [TextInputClient] usually maintain their own text editing states
+/// ([TextEditingValue]) separately. They must be kept in sync as long as the
+/// [TextInputClient] is connected. The following methods can be used to send
+/// [TextEditingValue] to update the other party, when either party's text
+/// editing states change:
+///
+/// * The [TextInput.attach] method allows a [TextInputClient] to establish a
+///   connection to the text input. An optional field in its `configuration`
+///   parameter can be used to specify an initial value for the system text
+///   input's [TextEditingValue].
+///
+/// * The [TextInputClient] sends its [TextEditingValue] to the system text
+///   input plugin using [TextInputConnection.setEditingState].
+///
+/// * The platform text input plugin sends its [TextEditingValue] to the
```

Comment:
Nit: I think that "platform" and "system" are being used interchangeably in these docs to refer to the engine side of things.  It think it would be useful to consolidate and use either one or the other exclusively.  Just a thought I had as I was reading through this and trying to follow, but up to you.

Reply:
Do we have a somewhat official name for the text input plugin? Or we just call it engine?

Reply:
I'm not sure...